pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +87,29 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func WaitAndRetryForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
```

Comment:
Since this function takes a ctx, it should abide by the context interface and cancel when the `context.CancelFunc` is called.  Just my opinion.